src/connectComp.cpp

```diff
@@ -0,0 +1,265 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <math.h>
+
+#include <vector>
+#include <iostream>
+
+#include <unordered_map>
+
+#include <unordered_set>
+#include <string.h>
+#include <cassert>
+#include <cstdlib>
+
+namespace found {
+
+struct Point {
+	int x;
+	int y;
+};
+
+struct Edge {
+	int magnitude; // number of points in this edge
+	std::vector<Point> points;
+	// int Xmin, Xmax, Ymin, Ymax;
+};
+
+std::vector<Edge> NaiveConnectedComponent(unsigned char *image, int imageWidth, int imageHeight) { // const??
+	int threshold = 3; // over threshold means possible components exist
```

Comment:
This is fine for now, I will put in an edit to this particular line when Karuna pushes her algorithm. In the meantime, please place a `TODO: Edit this for compatibility with edge filtering` comment above this